Cargo.toml

```diff
@@ -27,6 +27,7 @@ rayon = "1.5"
 num-traits = "0.2"
 num-bigint = "0.4"
 indexmap = "1.6.2"
+priority-queue = "1.1.1"
```

Comment:
I used priority_queue since i wanted to quickly change the priority of an element inside the queue (https://docs.rs/priority-queue/1.1.1/priority_queue/struct.PriorityQueue.html#method.change_priority_by) and `BinaryHeap` doesn't really offer this functionality. As for the general usage, i'm not 100% sure but i expect `BinaryHeap` to be faster (provided we only want to extract the max) as it's more lightweight: my understanding is that `PriorityQueue` is the same as a binary heap plus an extra `IndexMap` to store the position of an element in the heap.